File: Arrays/Algorithms/Rearrangement/rearrange_pos_neg.cpp
#include <bits/stdc++.h>
using namespace std;

void rearrangePosNeg(vector<int>& arr) {
	int n = arr.size();
	int i = -1;
	for (int j = 0; j < n; j++) {
		if (arr[j] < 0) swap(arr[++i], arr[j]);
	}
}

int main() {
	int n;
	cin >> n;
	vector<int> arr(n);
	for (int i = 0; i < n; i++) cin >> arr[i];
	rearrangePosNeg(arr);
	for (int x : arr) cout << x << ' ';
	cout << endl;
	return 0;
}

File: Arrays/Algorithms/Rearrangement/rearrange_pos_neg.py
def rearrange_pos_neg(arr):
	i = -1
	for j in range(len(arr)):
		if arr[j] < 0:
			i += 1
			arr[i], arr[j] = arr[j], arr[i]

if __name__ == "__main__":
	n = int(input())
	arr = list(map(int, input().split()))
	rearrange_pos_neg(arr)
	print(*arr)
